python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -604,18 +615,17 @@ def __init__(self):
         """
         super(GangaObject, self).__init__(None)
 
-        self._data_dict = {}
         self._index_cache_dict = {}
         self._registry = None
 
-        #Node.__init__(self, None)
-
         if self._schema is not None and hasattr(self._schema, 'allItems'):
+            self._data_dict = dict.fromkeys(self._schema.datadict)
             for attr, item in self._schema.allItems():
                 ## If an object is hidden behind a getter method we can't assign a parent or defvalue so don't bother - rcurrie
                 if item.getProperties()['getter'] is None:
                     setattr(self, attr, self._schema.getDefaultValue(attr))
-
+        else:
+            self._data_dict = {}
```

Comment:
I'm pretty sure that `self._schema` will never be `None` as it is always filled in by the metaclass. The whole `if` statement above is not needed as far as I can tell.